### DISCUSSION TITLE
Apply song select filters in main menu

### DISCUSSION THREAD

**Discussion by Moltanicaa** [ppy/osu#13140] [Ideas]
**Describe the new feature:** 
If I go into the game and I have a map in the main menu that does not have the difficulty of the current ruleset, then when I enter the list of cards, a completely different map will be displayed without the ability to return to the previous one.
**Proposal designs of the feature:** 
Maybe allow maps to be displayed without the current ruleset diff in the list? Or make a playlist only from the maps of the current ruleset?

https://youtu.be/nSRkFthZEtk

**Comment by bdach** [comment]
Allowing converted beatmaps to be displayed via settings is a workaround for this. Unsure if we want to move song select filter application out into a more global scope as it's quite brittle as-is already.

**Comment by Moltanicaa** [comment]
Then the option from the list with maps of only the current ruleset looks more logical.

**Comment by peppy** [comment]
I don't quite understand this. Is the issue the fact that the song changes and you don't want it to change?

The main menu is supposed to randomly play random songs for the purpose of not being silent. I'm not sure it should read into beatmaps futher than "do they have an audio track?".

**Comment by gagahpangeran** [comment]
The issue is if the song in the main menu is not from the chosen ruleset, when enter the song select, the song change because there is no beatmap with that song in that ruleset.

**Comment by peppy** [comment]
I already touched on that. Is that offensive to some?

**Comment by Moltanicaa** [comment]
The question here is whether there is logic in these actions. This situation would be logical if the main menu had a seasonal or default osu theme, but not a regular map. For new players, this can cause difficulties in understanding what is happening, because it is not entirely obvious why the current map is not displayed in the map list. The same situation will occur with the difficulty filter set.

**Comment by peppy** [comment]
So I can say for certain that we are not planning on filtering the main menu to current ruleset.

To the users in this thread that find this offensive: Would you prefer that your ruleset is changed when entering song select? That's the other option and I don't think it's what we want.

**Comment by Moltanicaa** [comment]
It was just a proposal, it can be closed, since I don't see much sense.
